aten/src/ATen/native/DistributionTemplates.h

```diff
@@ -124,7 +124,8 @@ at::Tensor& random_from_to_impl(at::Tensor& self, int64_t from, c10::optional<in
     int64_t to_inc = 0;
     if (isFloatingType(iter.dtype())) {
       AT_DISPATCH_FLOATING_TYPES_AND2(at::ScalarType::Half, at::ScalarType::BFloat16, self.scalar_type(), "random_from_to_range_calc", [&] {
-        to_inc = std::numeric_limits<scalar_t>::max() > std::numeric_limits<int64_t>::max() ? std::numeric_limits<int64_t>::max() : static_cast<int64_t>(std::numeric_limits<scalar_t>::max());
+        constexpr auto scalar_t_max = 1LL << std::numeric_limits<scalar_t>::digits;
```

Comment:
I guess you meant `int64_t` here (`LL` isn't necessarily `int64_t`)
```suggestion
        constexpr int64_t scalar_t_max = static_cast<int64_t>(1) << std::numeric_limits<scalar_t>::digits;
```

Reply:
xuhdev thanks!